### DISCUSSION TITLE
[CT-3018] [Feature] Add dependency graph to sources

### DISCUSSION THREAD

**Discussion by SeergeyL** [dbt-labs/dbt-core#8677] [Ideas]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Hi
I suggest adding a dependency graph feature to the sources in dbt. This enhancement would allow external models to be created by different services, and within dbt, models utilizing these sources would have a comprehensive dependency graph readily available.

### Describe alternatives you've considered

To achieve this, we can consider incorporating an additional optional yaml file containing dependencies for sources

### Who will this benefit?

This enhancement will particularly benefit users who have multiple services building models and wish to have a holistic view of the dependency graph within the models created by dbt.

### Are you interested in contributing this feature?

yes

**Comment by dbeatty10** [comment]
Thanks for this suggestion @SeergeyL 💡

I'm going to promote this to a Big Idea for further discussion.
